arrows/vxl/morphology.cxx

```diff
@@ -0,0 +1,325 @@
+
+  if( morphology_type != MORPHOLOGY_none )
+  {
+    for( auto i : vital::range::iota( input.nplanes() ) )
+    {
+      apply_morphology_to_plane( input, output, i );
+    }
+  }
+
+  if( combine_type == COMBINE_none )
+  {
+    // Don't combine across channels
+    return output;
+  }
+
+  // Select whether to do pixel-wise union or intersection
+  auto functor =
+    ( combine_type ==
+      COMBINE_union ) ? union_functor : intersection_functor;
+
+  auto accumulator = vil_plane( output, 0 );
+  for( unsigned i = 1; i < output.nplanes(); ++i )
+  {
+    auto current_plane = vil_plane( output, i );
+    // Union or intersect the current plane with the accumulator
+    vil_transform( accumulator, current_plane, accumulator, functor );
+  }
+  return accumulator;
+}
+
+// ----------------------------------------------------------------------------
+morphology
+::morphology()
+  : d{ new priv{} }
+{
+  attach_logger( "arrows.vxl.morphology" );
+}
+
+// ----------------------------------------------------------------------------
+morphology::~morphology() = default;
+
+// ----------------------------------------------------------------------------
+vital::config_block_sptr
+morphology
+::get_configuration() const
+{
+  // get base config from base class
+  vital::config_block_sptr config = algorithm::get_configuration();
+
+  config->set_value(
+    "morphology", morphology_converter().to_string( d->morphology_type ),
+    "Morphological operation to apply. Possible options are: " +
+    morphology_converter().element_name_string() );
+  config->set_value(
+    "element_shape", element_converter().to_string( d->element_type ),
+    "Shape of the structuring element. Possible options are: " +
+    element_converter().element_name_string() );
+  config->set_value( "element_size", d->element_size,
+                     "Size of the structuring element." );
+  config->set_value(
+    "channel_combination", combine_converter().to_string( d->combine_type ),
+    "Method for combining multiple binary channels. Possible options are: " +
+    morphology_converter().element_name_string() );
+
+  return config;
+}
+
+// ----------------------------------------------------------------------------
+void
+morphology
+::set_configuration( vital::config_block_sptr in_config )
+{
+  // Start with our generated vital::config_block to ensure that assumed values
+  // are present. An alternative would be to check for key presence before
+  // performing a get_value() call.
+  vital::config_block_sptr config = this->get_configuration();
+  config->merge_config( in_config );
+
+  d->morphology_type = config->get_enum_value< morphology_converter >(
+    "morphology" );
+  d->element_type = config->get_enum_value< element_converter >(
+    "element_shape" );
+  d->element_size = config->get_value< double >( "element_size" );
+  d->combine_type = config->get_enum_value< combine_converter >(
+    "channel_combination" );
```

Comment:
(Weak suggestion:)
```suggestion
  d->morphology_type =
    config->get_enum_value< morphology_converter >( "morphology" );
  d->element_type =
    config->get_enum_value< element_converter >( "element_shape" );
  d->element_size = config->get_value< double >( "element_size" );
  d->combine_type =
    config->get_enum_value< combine_converter >( "channel_combination" );
```

(If you think it looks better, you could also break the line that isn't broken for consistency, but either is fine.)